modules/xfem/src/base/XFEMFuncs.C

```diff
@@ -814,4 +814,131 @@ crossProduct2D(const Point & point_a, const Point & point_b)
   return (point_a(0) * point_b(1) - point_b(0) * point_a(1));
 }
 
+Real
+pointSegmentDistance(const Point & x0, const Point & x1, const Point & x2, Point & xp)
```

Comment:
I'm having a hard time understanding this fancy algorithm... Can we add some unit tests for this (and pointTriangleDistance)?